src/components/switch/switch.context.ts

```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+type SwitchContextType = {
+  loading?: boolean;
+};
+
+const SwitchContext = React.createContext<SwitchContextType | null>(null);
```

Comment:
```suggestion
const SwitchContext = React.createContext<SwitchContextType>({});
```

I don't see any reason to compicate matters by bringing `null` into it. The context type is an object with an (optional) boolean inside it, so you can make the default just an object that doesn't have it. (You could also do this with an explicit `{ loading: false }`, which would also allow you to make the type non-optional - but I don't care much whether you do that or not.)